src/Common/ThreadStatus.cpp

```
@@ -226,22 +224,12 @@ void ThreadStatus::attachInternalProfileEventsQueue(const InternalProfileEventsQ
 {
     profile_queue_ptr = profile_queue;
 
-    if (!thread_group)
-        return;
+    chassert(thread_group);
```

I would rather throw a logical error as well